File: lab2-functionalprimer/python3/func_as_arg.py
def listFunc():
    return [i for i in range(a, b)] #Create list of ints from a to b, Haskell equivalent [a..b]

def applicatorFunc(inpFunc, s):
    if s=='s':
        return sum(inpFunc())
    else:
        return sum(inpFunc())/5

a = int(input("Please enter a value for a"))
b = int(input("Please enter a value for b"))

print(applicatorFunc(listFunc, 's'))

File: lab2-functionalprimer/haskell/func_as_arg.hs
-- Generalized applicatorFunc
applicatorFunc :: [Int] -> Char -> Float
applicatorFunc xs s =
    if s == 's'
        then fromIntegral (sum xs)
        else fromIntegral (sum xs) / fromIntegral (length xs)

main = do
    putStrLn "Enter the start of the range (a):"
    aStr <- getLine
    putStrLn "Enter the end of the range (b):"
    bStr <- getLine
    putStrLn "Enter 's' for sum or anything else for average:"
    op <- getLine

    let a = read aStr :: Int
    let b = read bStr :: Int
    let inpList = [a..b]

    let result = applicatorFunc inpList (head op)
    putStrLn ("Result = " ++ show result)
